Sources/Lumina/Camera/Extensions/CaptureDeviceHandlerExtension.swift

```diff
@@ -0,0 +1,127 @@
+    for oldInput in self.session.inputs where oldInput == self.audioInput {
+      self.session.removeInput(oldInput)
+    }
+  }
+  
+  func purgeVideoDevices() {
+    LuminaLogger.notice(message: "purging old video devices on capture session")
+    for oldInput in self.session.inputs where oldInput == self.videoInput {
+      self.session.removeInput(oldInput)
+    }
+    for oldOutput in self.session.outputs {
+      if oldOutput == self.videoDataOutput || oldOutput == self.photoOutput || oldOutput == self.metadataOutput || oldOutput == self.videoFileOutput {
+        self.session.removeOutput(oldOutput)
+      }
+      if let dataOutput = oldOutput as? AVCaptureVideoDataOutput {
+        self.session.removeOutput(dataOutput)
+      }
+      if #available(iOS 11.0, *) {
+        if let depthOutput = oldOutput as? AVCaptureDepthDataOutput {
+          self.session.removeOutput(depthOutput)
+        }
+      }
+    }
+  }
+  
+  func getDevice(with position: AVCaptureDevice.Position) -> AVCaptureDevice? {
+#if swift(>=4.0.2)
+    if #available(iOS 11.1, *), position == .front {
+      if let device = AVCaptureDevice.default(.builtInTrueDepthCamera, for: .video, position: .front) {
+        self.currentCaptureDevice = device
+        return device
+      }
+    }
+#endif
+    if #available(iOS 10.2, *), let device = AVCaptureDevice.default(.builtInDualCamera, for: .video, position: position) {
+      self.currentCaptureDevice = device
+      return device
+    } else if let device = AVCaptureDevice.default(.builtInWideAngleCamera, for: .video, position: position) {
+      self.currentCaptureDevice = device
+      return device
+    }
+    return nil
+  }
+  
+  func configureFrameRate() {
+    guard let device = self.currentCaptureDevice else {
+      return
+    }
+    for vFormat in device.formats {
+      let dimensions = CMVideoFormatDescriptionGetDimensions(vFormat.formatDescription)
+      let ranges = vFormat.videoSupportedFrameRateRanges as [AVFrameRateRange]
+      guard let frameRate = ranges.first else {
+        continue
+      }
+      if frameRate.maxFrameRate >= Float64(self.frameRate) &&
+          frameRate.minFrameRate <= Float64(self.frameRate) &&
+          self.resolution.getDimensions().width == dimensions.width &&
+          self.resolution.getDimensions().height == dimensions.height &&
+          CMFormatDescriptionGetMediaSubType(vFormat.formatDescription) == 875704422 { // meant for full range 420f
+        do {
+          try device.lockForConfiguration()
+          device.activeFormat = vFormat as AVCaptureDevice.Format
+          device.activeVideoMinFrameDuration = CMTimeMake(value: 1, timescale: Int32(self.frameRate))
+          device.activeVideoMaxFrameDuration = CMTimeMake(value: 1, timescale: Int32(self.frameRate))
+          device.unlockForConfiguration()
+          break
+        } catch {
+          continue
+        }
+      }
+    }
+  }
+  
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/dokun1/Lumina/blob/5e323e0c2296f8f5c5b041c942e35a2720d9c81e/Sources/Lumina/Camera/Extensions/CaptureDeviceHandlerExtension.swift#L112"></span>Lines should not have trailing whitespace.<br />
<code>trailing_whitespace</code> <code>CaptureDeviceHandlerExtension.swift:112</code></td>
    </tr>
  </tbody>
</table>